generate_index_html.py

```diff
@@ -0,0 +1,148 @@
+import requests
+import re
+import subprocess
+import argparse
+from packaging.requirements import Requirement
+from packaging.version import Version
+
+
+def get_custom_released_package_versions(bucket: str, folder: str) -> dict:
+    data = subprocess.check_output([f"gsutil ls -lh 'gs://{bucket}/{folder}/' | grep -E '\.whl|\.tar\.gz'"], shell=True)
+    # Currently similar strings are correctly parsed:
+    # - 46.25 KiB  2022-07-20T21:18:06Z  gs://pip.astronomer.io/v2/apache-airflow-providers-cncf-kubernetes/apache_airflow_providers_cncf_kubernetes-1.0.2rc1-py3-none-any.whl
+    # - 46.25 KiB  2022-07-20T21:18:06Z  gs://pip.astronomer.io/v2/apache-airflow-providers-cncf-kubernetes/apache_airflow_providers_cncf_kubernetes-1.0.2+astro+1-py3-none-any.whl
+    # - 46.25 KiB  2022-07-20T21:18:06Z  gs://pip.astronomer.io/v2/apache-airflow-providers-cncf-kubernetes/apache_airflow_providers_cncf_kubernetes-1.0.2-py3-none-any.tar.gz
+    # - 46.25 KiB  2022-07-20T21:18:06Z  gs://pip.astronomer.io/v2/apache-airflow-providers-cncf-kubernetes/apache_airflow_providers_cncf_kubernetes-1.0.2-py3-none-any.whl
+    # Note - platform information is not handled atm.
+
+    # For example if the input string is -
+    # 46.25 KiB  2022-07-20T21:18:06Z  gs://pip.astronomer.io/v2/apache-airflow-providers-cncf-kubernetes/apache_airflow_providers_cncf_kubernetes-1.0.2rc1-py3-none-any.whl
+    # The regex match groups will be
+    # Group 0: 46.25 KiB
+    # Group 1: 2022-07-20T21:18:06Z
+    # Group 2: pip.astronomer.io
+    # Group 3: v2
+    # Group 4: apache-airflow-providers-cncf-kubernetes/apache_airflow_providers_cncf_kubernetes-1.0.2rc1-py3-none-any.whl
+    # Group 5: apache-airflow-providers-cncf-kubernetes/apache_airflow_providers_cncf_kubernetes
+    # Group 6: 2.0.2
+
+    pattern = re.compile("([\w\W]+?)  ([\w\W]+?)  gs://([\w\W]+?(?=/))/([\w\W]+?(?=/))/(([\w\W]+?)-(\d+\.\d+\.\d+(rc\d+)?(\+astro\.\d+)?)-py3-none-any(?:\.whl|\.tar\.gz))")
+    matches = pattern.findall(data.decode())
+    version_to_files = {}
+    # Example -
+    # version_to_files = {
+    #   "2.0.2": (
+    #       "46.25 KiB",
+    #       "2022-07-20T21:18:06Z",
+    #       "apache-airflow-providers-cncf-kubernetes/apache_airflow_providers_cncf_kubernetes-1.0.2rc1-py3-none-any.whl"
+    #   )
+    # }
+    for match in matches:
+        if match[6] not in version_to_files:
+            version_to_files[match[6]] = []
+        version_to_files[match[6]].append((match[0], match[1], match[4]))
+    return version_to_files
+
+
+def get_released_public_package_list(package: str) -> dict:
+    url = f"https://pypi.org/pypi/{package}/json"
+    response = requests.get(url)
+    response.raise_for_status()
+    return {release: files for release, files in response.json()["releases"].items()}
```

Comment:
```suggestion
    return {response.json()["releases"]}
```

No need to iterate through them all, right?